--- build.rs
+++ build.rs
@@ -65,8 +65,11 @@
     match platform {
         Platform::Desktop => conf.define("PLATFORM", "Desktop"),
         Platform::Web => conf.define("PLATFORM", "Web"),
         Platform::RPI => conf.define("PLATFORM", "Raspberry Pi"),
     };
+    if platform_os == PlatformOS::Linux && platform != Platform::RPI {
+        conf.define("GRAPHICS", "GRAPHICS_API_OPENGL_21");
+    }

     let dst = conf.build();
     let dst_lib = join_cmake_lib_directory(dst);
@@ -172,7 +172,8 @@
             println!("cargo:rustc-link-lib=dylib=shell32");
         }
         PlatformOS::Linux => {
             println!("cargo:rustc-link-search=/usr/local/lib");
+            println!("cargo:rustc-link-search=/usr/lib/arm-linux-gnueabihf");
             println!("cargo:rustc-link-lib=X11");
         }
         PlatformOS::OSX => {
@@ -188,7 +188,8 @@
     }
     if platform == Platform::Web {
         println!("cargo:rustc-link-lib=glfw");
-    }
+    } else if platform == Platform::RPI {
+        println!("cargo:rustc-link-search=/opt/vc/lib");
+        println!("cargo:rustc-link-lib=bcm_host");
+        println!("cargo:rustc-link-lib=brcmEGL");
+        println!("cargo:rustc-link-lib=brcmGLESv2");
+        println!("cargo:rustc-link-lib=vcos");
+    }
 
     println!("cargo:rustc-link-lib=static=raylib");
 }
@@ -245,7 +245,7 @@
         // Cargo web takes care of this but better safe than sorry
         env::set_var("EMMAKEN_CFLAGS", "-s USE_GLFW=3");
         Platform::Web
-    } else if target.contains("armv7-unknown-linux") {
+    } else if target.contains("armv7-unknown-linux") || target.contains("arm-unknown-linux") {
         Platform::RPI
     } else {
         Platform::Desktop
--- raylib/src/config.h
+++ raylib/src/config.h
@@ -125,6 +125,6 @@
 //#define SUPPORT_FILEFORMAT_BMP      1
 //#define SUPPORT_FILEFORMAT_TGA      1
-//#define SUPPORT_FILEFORMAT_JPG      1
+#define SUPPORT_FILEFORMAT_JPG      1
 #define SUPPORT_FILEFORMAT_GIF      1
 //#define SUPPORT_FILEFORMAT_PSD      1
 #define SUPPORT_FILEFORMAT_DDS      1
--- raylib/src/core.c
+++ raylib/src/core.c
@@ -4354,7 +4354,11 @@
     // Load OpenGL extensions
     // NOTE: GLFW loader function is required by GLAD but only used for OpenGL 2.1 and 3.3,
     // OpenGL ES 2.0 extensions (and entry points) are loaded manually using eglGetProcAddress()
-    rlLoadExtensions(glfwGetProcAddress);
+#if defined(PLATFORM_DESKTOP) || defined(PLATFORM_WEB)
+    rlLoadExtensions(glfwGetProcAddress);
+#else
+    rlLoadExtensions(eglGetProcAddress);
+#endif
 
     // Initialize OpenGL context (states and resources)
     // NOTE: CORE.Window.screen.width and CORE.Window.screen.height not used, just stored as globals in rlgl
